### PR TITLE
add serp specific variants and use weighted approach for distribution

### PR DESCRIPTION
**Pull request by brindy**
Task/Issue URL: https://app.asana.com/0/414235014887631/757819156125526
Tech Design URL:
CC:

**Description**:

Add 'sc' and 'sd' variants to the app for use by the SERP.

**Steps to test this PR**:

1. Launch the app and ensure allocation to one of sc or sd

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

File: Core/VariantManager.swift [hunk 1]
```diff
@@ -104,8 +108,11 @@ public class Arc4RandomUniformVariantRNG: VariantRNG {
 
     public init() { }
 
-    public func nextInt(upperBound: Int) -> Int {
-        return Int(arc4random_uniform(UInt32(upperBound)))
```
Comment (subsymbolic) [inline Core/VariantManager.swift]: Why change this? Was there a problem with using arc4random_uniform? I would expect it to be fine for our needs. Looking forward to Swift 4.2 new random methods soon!

File: Core/VariantManager.swift [hunk 2]
```diff
@@ -28,18 +28,21 @@ public enum FeatureName {
 public struct Variant {
 
     public static let defaultVariants: [Variant] = [
-        // None
+        
+        // For the SERP - do not remove:
+        Variant(name: "sc", weight: 1.0, features: []),
```
Comment (subsymbolic) [inline Core/VariantManager.swift]: How would you feel about using ints for the weights? It feels like overkill to allow both arbitrary values and allow them to be doubles.

**Comment by brindy** [comment]
@subsymbolic I reset it and reimplemented it using ints and the original arc4random_uniform function - I did a force push, so please check it out fresh

**Review by subsymbolic — APPROVED** [review]
Looks good!

**Pull request merged**